core/events/__init__.py

```diff
@@ -69,8 +68,7 @@
 
    >>> user = events.User(123, "joe@bloggs.com")
    >>> creation = events.CreateSubmission(creator=user)
-   >>> metadata = [('title', 'A new theory of foo')]
-   >>> update = events.UpdateMetadata(creator=user, metadata=metadata)
+   >>> update = events.SetTitle(creator=user, title='A new theory of foo')
    >>> submission = events.save(creation, update)
```

Comment:
This example seems like it wouldn't work as intended - the SetTitle event should require a `submission_id` which can't be extracted until the `CreateSubmission` event is committed. Perhaps my understanding of the events model is wrong though.

Reply:
The event can be instantiated before there is a submission. But both the ``project`` and ``validate`` methods on the event require a submission. That's the glue that ``save`` provides. Note that ``save`` will internally pick up the submission ID from the creation event, so it need not be passed.